consensus/src/round_manager.rs

```diff
@@ -478,6 +478,21 @@ impl RoundManager {
 
         let block_time_since_epoch = Duration::from_micros(proposal.timestamp_usecs());
 
+        // If the timestamp is too far in the future, we log it as a security issue
+        if let Some(time_to_deadline) =
+            block_time_since_epoch.checked_sub(self.round_state.current_round_deadline())
+        {
+            // We can say that "too far" means more than 5 rounds ahead
+            if time_to_deadline > self.round_state.get_time_until_round(5) {
```

Comment:
can we move this to the block.verify function? I think local_timestamp + X sec is probably good enough too

Reply:
I think we can check if a block is too far in the future in `block.verify()`, but what about checking if a proposal is too far in the past? That also shouldn't happen.

Reply:
Sure. Do you mean the verify() function in consensus/consensus-types/src/proposal_msg.rs or should I maybe log this in the verify_well_formed function instead?

https://github.com/libra/libra/blob/3bf241dd282c8c245da68efc259c11ff79e4a28c/consensus/consensus-types/src/proposal_msg.rs#L32

What should X be for it to be probably malicious/worth reporting? Opinions @mimoo ?

Reply:
my understanding is that If you verify in `proposal_msg` then you won't caught the problem when you get a block through the block request API, instead [block.verify()](https://github.com/libra/libra/blob/cdbf57301aaee34d6b0102918a87826669e5917d/consensus/consensus-types/src/block.rs#L214) is a better place because it should be call by both APIs (it is called by proposal_msg.verify() [here](https://github.com/libra/libra/blob/36ca5cdeefabd54509f428c418ebfdf7d2186f10/consensus/consensus-types/src/proposal_msg.rs#L39)).

But then, anything that doesn't pass `verify()` is [security-logged anyway](https://github.com/libra/libra/blob/d024369f7347570f7b9953c2cb5609af638e8c7d/consensus/src/epoch_manager.rs#L390), so we might want to just add an actual check (like all the other checks implemented in `verify()`) that the block is not too far in the future.

Reply:
Done. This makes the change much smaller and contained too.
I arbitrarily set X to 5 minutes.